test/python/circuit/test_controlled_gate.py

```diff
@@ -355,9 +483,97 @@ def test_rotation_gates(self):
+
+            # call mcrx/mcry/mcrz
+            if base_gate_name == 'y':
+                qc.mcry(theta, q_controls, q_target[0], None, mode='noancilla',
+                        use_basis_gates=use_basis_gates)
+            else:  # case 'x' or 'z' only support the noancilla mode and do not have this keyword
+                getattr(qc, 'mcr' + base_gate_name)(theta, q_controls, q_target[0],
+                                                    use_basis_gates=use_basis_gates)
+
+            for idx, bit in enumerate(bitstr):
+                if bit == '0':
+                    qc.x(q_controls[idx])
+
+            backend = BasicAer.get_backend('unitary_simulator')
+            simulated = execute(qc, backend).result().get_unitary(qc)
+
+            if base_gate_name == 'x':
+                rot_mat = RXGate(theta).to_matrix()
+            elif base_gate_name == 'y':
+                rot_mat = RYGate(theta).to_matrix()
+            else:  # case 'z'
+                rot_mat = U1Gate(theta).to_matrix()
+
+            expected = _compute_control_matrix(rot_mat, num_controls, ctrl_state=ctrl_state)
+            self.assertTrue(matrix_equal(simulated, expected))
+
+    @combine(num_controls=[1, 2, 4], use_basis_gates=[True, False])
+    def test_multi_controlled_y_rotation_matrix_basic_mode(self, num_controls, use_basis_gates):
+        """Test the multi controlled Y rotation using the mode 'basic'."""
+
+        # get the number of required ancilla qubits
+        if num_controls <= 2:
+            num_ancillas = 0
+        else:
+            num_ancillas = num_controls - 2
+
+        q_controls = QuantumRegister(num_controls)
+        q_target = QuantumRegister(1)
+
+        for ctrl_state in range(2**num_controls):
+            bitstr = bin(ctrl_state)[2:].zfill(num_controls)[::-1]
+            theta = 0.871236 * pi
+            if num_ancillas > 0:
+                q_ancillas = QuantumRegister(num_ancillas)
+                qc = QuantumCircuit(q_controls, q_target, q_ancillas)
+            else:
+                qc = QuantumCircuit(q_controls, q_target)
+                q_ancillas = None
+
+            for idx, bit in enumerate(bitstr):
+                if bit == '0':
+                    qc.x(q_controls[idx])
+
+            qc.mcry(theta, q_controls, q_target[0], q_ancillas, mode='basic',
+                    use_basis_gates=use_basis_gates)
+
+            for idx, bit in enumerate(bitstr):
+                if bit == '0':
+                    qc.x(q_controls[idx])
+
+            rot_mat = RYGate(theta).to_matrix()
+
+            backend = BasicAer.get_backend('unitary_simulator')
+            simulated = execute(qc, backend).result().get_unitary(qc)
+            if num_ancillas > 0:
+                simulated = simulated[:2**(num_controls + 1), :2**(num_controls+1)]
+
+            expected = _compute_control_matrix(rot_mat, num_controls, ctrl_state=ctrl_state)
+
+            self.assertTrue(matrix_equal(simulated, expected))
```

Comment:
Same comment as in https://github.com/Qiskit/qiskit-terra/pull/3714/files#r382785476